packages/labs/src/components/select/suggest.tsx

```diff
@@ -0,0 +1,296 @@
+
+        return (
+            <Popover
+                autoFocus={false}
+                enforceFocus={false}
+                isOpen={this.state.isOpen}
+                position={Position.BOTTOM_LEFT}
+                {...popoverProps}
+                className={classNames(listProps.className, popoverProps.className)}
+                onInteraction={this.handlePopoverInteraction}
+                popoverClassName={classNames(Classes.SELECT_POPOVER, popoverProps.popoverClassName)}
+                popoverDidOpen={this.handlePopoverDidOpen}
+                popoverWillClose={this.handlePopoverWillClose}
+            >
+                <div
+                    onKeyDown={this.getTargetKeyDownHandler(handleKeyDown)}
+                    onKeyUp={this.state.isOpen ? handleKeyUp : undefined}
+                >
+                    <InputGroup
+                        placeholder="Search..."
+                        value={inputValue}
+                        {...htmlInputProps}
+                        inputRef={this.refHandlers.input}
+                        onChange={this.handleQueryChange}
+                        onFocus={this.handleInputFocus}
+                    />
+                </div>
+                <div onKeyDown={handleKeyDown} onKeyUp={handleKeyUp}>
+                    <Menu ulRef={listProps.itemsParentRef}>
+                        {this.renderItems(listProps)}
+                    </Menu>
+                </div>
+            </Popover>
+        );
+    }
+
+    private renderItems({ activeItem, filteredItems, handleItemSelect }: IQueryListRendererProps<T>) {
+        const { itemRenderer, noResults } = this.props;
+        if (filteredItems.length === 0) {
+            return noResults;
+        }
+        return filteredItems.map((item, index) => itemRenderer({
+            index,
+            item,
+            handleClick: (e) => handleItemSelect(item, e),
+            isActive: item === activeItem,
+        }));
+    }
+
+    private selectText = () => {
+        if (this.input != null) {
+            // wait until the input is properly focused to select the text inside of it
+            requestAnimationFrame(() => this.input.setSelectionRange(0, this.input.value.length));
+        }
+    }
+
+    private handleInputFocus = (event: React.SyntheticEvent<HTMLInputElement>) => {
+        const { openOnKeyDown, inputProps = {} } = this.props;
+
+        this.selectText();
+
+        if (!openOnKeyDown) {
+            this.setState({ isOpen: true });
+        }
+
+        Utils.safeInvoke(inputProps.onFocus, event);
+    }
+
+    private handleActiveItemChange = (activeItem: T) => this.setState({ activeItem });
+
+    private handleItemSelect = (item: T, event: React.SyntheticEvent<HTMLElement>) => {
+        const { closeOnSelect } = this.props;
+        let nextOpenState: boolean;
+
+        if (!closeOnSelect) {
+            this.input.focus();
+            this.selectText();
```

Comment:
Is this line necessary even though `focus`'ing will trigger `handleInputFocus` and subsequently invoke `this.selectText()`? (Per the `requestAnimationFrame` comment above, I'm assuming focus doesn't happen synchronously here?)